test/accountSpec.js

```diff
@@ -689,6 +686,88 @@ describe('Accounts', function () {
       })
     })
 
+    it('should resend notifications when asked to', function * () {
+      const listener = sinon.spy()
+      this.socket.on('message', (msg) => listener(JSON.parse(msg)))
+
+      const transfer = this.transfer
+
+      yield this.request()
+        .put(transfer.id)
+        .auth('alice', 'alice')
+        .send(transfer)
+        .expect(201)
+        .expect(validator.validateTransfer)
+        .end()
+
+      // TODO: Is there a more elegant way?
+      yield timingHelper.sleep(50)
+
+      const transferId = transfer.id.match('http://localhost/transfers/(.+)')[1]
+      this.socket.send(JSON.stringify({ type: 'request_notification',
+                                        id: transferId }))
+
+      yield timingHelper.sleep(50)
+
+      sinon.assert.calledTwice(listener)
```

Comment:
For (2), how would you make sure a transfer is already in the db? If it is by doing a PUT then it looks like (1) would be a subset of (2) and therefore isn't necessary?